### PR TITLE
demonstration of optimizer-style learning rate

### PR DESCRIPTION
**Pull request by zcain117**
Showing option #2 for the learning rate scheduling that we need for resnet50 to pass 76% accuracy.

I will also work on a PR for option #3, which is to subclass [torch.optim.lr_scheduler](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html). At first I was hesitant since our new scheduler depends on both epoch and step within an epoch, so it breaks the [Pytorch paradigm for how they intend schedulers to be used](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate).

However, I think that option #3 will be require less code in test_train_imagenet.py, so I will try to implement that so everyone can see the other option before we decide on a which strategy we want to submit

### CODE REVIEW

**Comment by zcain117** [comment]
Added the scheduler-style option here: https://github.com/pytorch/xla/pull/1012

File: test/optimizers.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import test_utils
+from torch.optim import SGD
+
+MIN_DELTA_TO_UPDATE_LR = 0.000001
+NUM_WARMUP_EPOCHS = 1
+
+
+class WarmupAndExponentialDecaySGD(SGD):
```
Comment (dlibenzi) [inline test/optimizers.py]: I think this might be a good approach, though this should be a wrapper optimizer.
It should take an optimizer as constructor argument, and not subclass SGD.
Otherwise if you want to add LR schedule to another optimizer class, you will have to duplicate the work.
